subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/JvmTestSuitePluginIntegrationTest.groovy

```diff
@@ -0,0 +1,267 @@
+        def buildFileA = subADir.file("build.gradle") << """
+            plugins {
+                id 'jvm-test-suite'
+                id 'java'
+            }
+
+            repositories {
+                ${mavenCentralRepository()}
+            }
+
+            testing {
+                suites {
+                    test {
+                        useJUnit()
+                        dependencies {
+                            implementation project
+                        }
+                    }
+                }
+            }
+            """.stripIndent()
+
+        file("src/test/java/SomeTestA.java") << """
+            import org.junit.Test;
+
+            public class SomeTestA {
+                @Test public void foo() {}
+            }
+            """.stripIndent()
+
+        def subBDir = createDir("subB")
+        def buildFileB = subBDir.file("build.gradle") << """
+            plugins {
+                id 'jvm-test-suite'
+                id 'java'
+            }
+
+            repositories {
+                ${mavenCentralRepository()}
+            }
+
+            testing {
+                suites {
+                    test {
+                        useJUnit()
+                        dependencies {
+                            implementation project
+                        }
+                    }
+                }
+            }
+            """.stripIndent()
+
+        file("src/test/java/SomeTestB.java") << """
+            import org.junit.Test;
+
+            public class SomeTestB {
+                @Test public void foo() {}
+            }
+            """.stripIndent()
+
+        settingsFile << """
+            include ':subA'
+            include ':subB'
+            """.stripIndent()
+
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
+            dependencies {
+                implementation project(':subA')
+                implementation project(':subB')
+            }
+
+            // A resolvable configuration to collect test results data
+            def testDataConfig = configurations.create("testData") {
+                visible = false
+                canBeResolved = true
+                canBeConsumed = false
+                extendsFrom(configurations.implementation)
+                attributes {
+                    attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage, Usage.VERIFICATION))
+                    attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category, Category.DOCUMENTATION))
+                    attribute(DocsType.DOCS_TYPE_ATTRIBUTE, objects.named(DocsType, DocsType.TEST_RESULTS))
+                }
+            }
+
+            def testResolve = tasks.register('testResolve') {
+                doLast {
+                    assert testDataConfig.getResolvedConfiguration().getFiles().containsAll([project(':subA').tasks["test"].binaryResultsDirectory.file("results.bin").get().getAsFile(),
```

Comment:
instead of reaching into other projects, it might be easier here to just manually specify the path.  e.g., just use `file("subA/build/test/results/results.bin")` (or whatever the right path is).  That'll let this test run with configuration caching and it's unlikely that we're going to be moving binaryResultsDirectory around in a way that makes this harder to maintain.